### PR TITLE
Track pageviews in Posthog under a new 'Docs' source

### PR DESCRIPTION
**Pull request by esthersweon**
### Public-Facing Changes

- N/A

### Description

- Track pageviews in Posthog under a new 'Docs' source

### CODE REVIEW

File: docusaurusConfig.ts [hunk 1]
```diff
@@ -86,6 +86,17 @@ const docusaurusConfig: Config = {
...
       } satisfies import("@docusaurus/plugin-client-redirects").Options,
     ],
+    [
+      "posthog-docusaurus",
+      {
+        apiKey: process.env.POSTHOG_KEY,
+        appUrl: "https://docs.foxglove.dev",
+        enableInDevelopment: false,
+        sanitize_properties: (properties: Record<string, unknown>): Record<string, unknown> => {
+          return { ...properties, Source: "Docs" };
```
Comment (amacneil) [inline docusaurusConfig.ts:96]:
```suggestion
          return { ...properties, Platform: "Web", Source: "Docs", };
```

**Comment by defunctzombie** [comment]
@amacneil do we still want this?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -10,6 +10,7 @@ jobs:
   all:
     name: all
     runs-on: ubuntu-latest
+    environment: ${{ github.ref == 'refs/heads/main' && 'foxglove-docs (Production)' || 'foxglove-docs (Preview)' }}
```
Comment (amacneil) [inline .github/workflows/ci.yml:13]: Is this change related?

**Comment by amacneil** [comment]
Yes we definitely want this. I have not tested this PR though.
